### PR TITLE
Jseldess/faq

### PR DESCRIPTION
**Pull request by jseldess**
I think our faq is close-to-final for beta, but I want final opinions and feedback. I addressed @vivekmenezes' and most of @tschottdorf's feedback in issue #8. Some open questions:

- Under **Does CockroachDB have ACID semantics?**, Toby gave the following feedback:
  - Consistency: I think this point should focus on our optimistic concurrency model. Raft is really more about keeping consistent replicas (which is another meaning of "consistent")
  - Durability: again there is a mixup of Raft and Replication which I think could be clarified.

@tschottdorf, can you suggest some rewrites here?

@mberhault, please take a look at my answer for **What is CockroachDB’s security model?**.

cc: @bdarnell, @petermattis, @jess-edwards, @mrtracy

### CODE REVIEW

File: frequently-asked-questions.md [hunk 1]
```diff
@@ -121,10 +129,17 @@ CockroachDB is not yet suitable for real-time analytics, although support for an
 
 ## What is CockroachDB’s security model?
 
-TBD. We support client and internode SSL. Everything within CockroachDB requires that you speak to it with encryption. 
+You can run a secure or insecure CockroachDB cluster. When secure, client/node and inter-node communication is encrypted, and SSL certificates authenticate the identity of both clients and nodes. When insecure, there's no encryption or authentication.
+
+Also, CockroachDB supports common SQL privileges on databases and tables. The `root` user has privileges for all databases, while unique users can be granted privileges for specific statements on the database and table levels.
```
Comment (mberhault) [inline frequently-asked-questions.md]: "statement at the database level" rather than "on the database level"

**Comment by mberhault** [comment]
LGTM, specifically for the security model section.

**Comment by jseldess** [comment]
Thanks, @mberhault. Changed that bit of wording in the security faq.

File: frequently-asked-questions.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ CockroachDB is especially well suited for applications that require:
 
 ## How does CockroachDB scale?
 
-CockroachDB scales horizontally with minimal operator overhead. You can run it on your local computer, a single server, a corporate development cluster, or the private or public cloud. Adding capacity is as easy as pointing a new node at the running cluster. 
+CockroachDB scales horizontally with minimal operator overhead. You can run it on your local computer, a single server, a corporate development cluster, or the private or public cloud. [Adding capacity](start-a-node.html) is as easy as pointing a new node at the running cluster.
```
Comment (petermattis) [inline frequently-asked-questions.md]: I think this should read `or a private or public cloud` as there isn't one specific private or public cloud.

File: frequently-asked-questions.md [hunk 3]
```diff
@@ -41,17 +41,17 @@ CockroachDB replicates your data for availability and guarantees consistency bet
...
+For short-term failures, such as a server restart, CockroachDB uses Raft to continue seamlessly as long as a majority of replicas remain available. Raft makes sure that a new “leader” for each group of replicas is elected if the former leader fails, so that subsequent transactions can continue and affected replicas can rejoin their group once they’re back online. For longer-term failures, such as a server/rack going down for an extended period of time or a datacenter outage, CockroachDB automatically rebalances replicas from the missing nodes, using the unaffected replicas as sources. Using capacity information from the gossip network, new locations in the cluster are identified and the missing replicas are re-replicated in a distributed fashion using all available nodes and the aggregate disk and network bandwidth of the cluster.
```
Comment (petermattis) [inline frequently-asked-questions.md]: The use of `subsequent transactions` makes me think that failure of a leader causes existing transactions to fail, but that isn't quite correct. When a node fails, only transactions that were being coordinated by that node will definitely fail. Others should experience a hiccup. I think you should drop the term `subsequent`.

**Comment by petermattis** [comment]
LGTM

File: frequently-asked-questions.md (hunk 2 above)
Comment (jseldess) [inline frequently-asked-questions.md]: Fixed with next commit.

File: frequently-asked-questions.md (hunk 3 above)
Comment (jseldess) [inline frequently-asked-questions.md]: Fixed with next commit. Thanks, @petermattis.

**Comment by petermattis** [comment]
LGTM

File: frequently-asked-questions.md [hunk 4]
```diff
@@ -87,22 +87,29 @@ Cockroach supports the PostgreSQL wire protocol, so you can use any available Po
...
 
 While all of these are distributed databases, only CockroachDB supports distributed transactions and provides strong consistency. Also, these other databases provide custom APIs, whereas CockroachDB offers standard SQL with extensions. 
 
+For more insight, see [CockroachDB in Comparison](cockroachdb-in-comparison.html).
+
 ## Can a MySQL or Postgres application be migrated to CockroachDB?
 
-The Alpha (soon Beta) version of CockroachDB is intended for use with new applications. The initial subset of SQL we support is small relative to the extensive standard, and every popular database implements its own set of extensions and exhibits a unique set of idiosyncracies. This makes porting an existing application impractical unless it is only a very lightweight consumer of SQL functionality.
+The current version of CockroachDB is intended for use with new applications. The initial subset of SQL we support is small relative to the extensive standard, and every popular database implements its own set of extensions and exhibits a unique set of idiosyncracies. This makes porting an existing application impractical unless it is only a very lightweight consumer of SQL functionality.
```
Comment (spencerkimball) [inline frequently-asked-questions.md]: s/impractical/non-trivial/

File: frequently-asked-questions.md [hunk 5]
```diff
@@ -121,10 +129,17 @@ CockroachDB is not yet suitable for real-time analytics, although support for an
```
Comment (spencerkimball) [inline frequently-asked-questions.md]: You should also mention use cases which require joins in SQL (but qualify that by saying they're slated for the next major release). I would also say that CockroachDB is not a good choice when very low latency reads and writes are critical; use an in-memory database instead.

**Comment by spencerkimball** [comment]
LGTM!

File: frequently-asked-questions.md (hunk 4 above)
Comment (jseldess) [inline frequently-asked-questions.md]: Will be changed with next commit.

File: frequently-asked-questions.md (hunk 5 above)
Comment (jseldess) [inline frequently-asked-questions.md]: Extending this with next commit. Thanks, @spencerkimball.

**Pull request merged**